engine-nextgen/src/main/kotlin/graphql/nadel/validation/NadelFieldValidation.kt

```diff
@@ -99,14 +102,23 @@ internal class NadelFieldValidation(
                     MissingArgumentOnUnderlying(parent, overallField, underlyingField, overallArg),
                 )
             } else {
-                // TODO check the type wrappings are equal
-                typeValidation.validate(
-                    NadelServiceSchemaElement(
-                        service = parent.service,
-                        overall = overallArg.type.unwrapAll(),
-                        underlying = underlyingArg.type.unwrapAll(),
-                    )
+                val unwrappedTypeIssues = typeValidation.validate(
+                        NadelServiceSchemaElement(
+                                service = parent.service,
+                                overall = overallArg.type.unwrapAll(),
+                                underlying = underlyingArg.type.unwrapAll(),
+                        )
                 )
+
+                val inputTypeIssues = if (!isMatchingInputTypeNameIgnoringNullableRenameScalar(overallArg.type, underlyingArg.type)) {
```

Comment:
This should be `val inputType = typeValidation.validateInputType` for code consistency. The caller shouldn't have to generate the error.